aopy/analysis.py

```diff
@@ -1543,3 +1543,61 @@ def interp_multichannel(x):
     x[nan_idx] = np.interp(idx,xp,fp)
 
     return x
+
+
+'''
+Behavioral metrics 
+'''
+
+def compute_path_length_per_trial(trajectory):
+    '''
+    This function calculates the path length by computing the distance from all points in the trajectory. It returns a single value for path length.
+
+    Args:
+        trajectory (nt x 2): Input trajectory, could be a corsor trajectory or eye trajectory
+
+    Returns:
+        path_length (float): length of the trajectory
+    '''
+    lengths = np.sqrt(np.sum(np.diff(trajectory, axis=0)**2, axis=1)) # compute the distance from all points in trajectory
+    path_length = np.sum(lengths)
+    return path_length
+
+
+def time_to_target(event_codes, event_times, per_target_stats=False):
+    '''
+    This function calculates reach time to target only on rewarded trials given event codes and event times.
+
+    Args:
+        event_codes (list) : event codes
+        event_times (list) : event times corresponding to the event codes. These event codes and event times could be the output of preproc.base.get_trial_segments_and_times().
+         get_stats_per_target (bool): optional, use if you want to calculate reach time per target
+
+    Returns:
+        reach_times (list):
+        reach_times_per_target(list of lists) : optional if per_target_stats == True
+    '''
+    CURSOR_ENTER_PERIPHERAL_TARGET = list(range(81, 89))
+    CENTER_TARGET_OFF = 32
+    REWARD = 48
+    tr_e = np.array([event_codes[iTr] for iTr in range(len(event_times)) if REWARD in event_codes[iTr]])
+    tr_t = np.array([event_times[iTr] for iTr in range(len(event_times)) if REWARD in event_codes[iTr]])
```

Comment:
why is it necessary to restrict to rewarded trials? maybe leave this up to the user?